### DISCUSSION TITLE
[RouteContext] Global state manage

### DISCUSSION THREAD

**Discussion by sendya** [vueComponent/pro-layout#96] [Ideas]
背景：
- React antd Pro v5 中，利用了 React.Context 来创建全局共享的状态管理，用来同步嵌套组件下的一些状态控制。

现状：
- Vue3 Pro 版本中，我利用 provide, inject 来进行全局的状态控制。

----

但目前的使用上，感觉体验并不是很好
- 从 Vue-Devtools 中查看，嵌套层级过多，有 RouteContextProvider，ProProvider 
- 使用上，Vue3 的 `reactive` 创建的响应式数据和 Vuex 不太好配合使用，但目前大部分人使用 Vue 时，还是会利用 Vuex 来储存和共享一些系统配置。

一些思考：
- RouteContextProvider，ProProvider 是否能合并成一个？
- RouteContext 的状态怎么管理比较好？

**Comment by lihxhit** [comment]
虽然一些实现复杂度上不是很清楚，但还是希望能与react版的pro-layout API与代码逻辑保持一致

**Reply by hanrea** [reply]
这个似乎没必要，选择vue就不太会选择react了。已经是vue的分支方向何必靠拢react？
